### PR TITLE
IP-1929 - Migrate RD Clinical Report from 6 to 5

### PR DESCRIPTION
**Pull request by greglever**
[IP-1929 - Migrate RD Clinical Report from 6 to 5](https://jira.extge.co.uk/browse/IP-1929)
====================

### CODE REVIEW

**Comment by greglever** [comment]
@priesgo @kevinpetersavage -- this is now rebased with IP-1913 and ready for review

File: protocols/migration/migration_reports_6_0_0_to_reports_5_0_0.py [hunk 1]
```diff
@@ -35,11 +35,7 @@ def migrate_interpreted_genome_to_interpreted_genome_rd(self, old_instance):
         return self.validate_object(object_to_validate=new_instance, object_type=self.new_model.InterpretedGenomeRD)
 
     def migrate_small_variants_to_reported_variants(self, small_variants):
-        if small_variants is None:
-            msg = "Small Variants are required to migrate from InterpretedGenome version 6 to InterpretedGenomeRD "
-            msg += "version 5 as Reported Variants are required"
-            raise MigrationError(msg)
-        return [
+        return [] if small_variants is None else [
```
Comment (priesgo) [inline protocols/migration/migration_reports_6_0_0_to_reports_5_0_0.py]: :+1:

File: protocols/migration/migration_reports_6_0_0_to_reports_5_0_0.py [hunk 2]
```diff
@@ -129,9 +125,13 @@ def migrate_report_event(self, old_event):
...
 
     def migrate_gene_panel(self, old_panel):
-        if old_panel is None or old_panel.panelName is None:
+        if old_panel is None:
             return None
         new_instance = self.convert_class(target_klass=self.new_model.GenePanel, instance=old_panel)
+        if old_panel.panelName is None:
+            new_instance.panelName = ""
+        if old_panel.panelVersion is None:
+            new_instance.panelVersion = ""
```
Comment (greglever) [inline protocols/migration/migration_reports_6_0_0_to_reports_5_0_0.py]: happier @priesgo ?

File: protocols/tests/test_migration/test_migration_reports_6_0_0_to_reports_5_0_0.py [hunk 3]
```diff
@@ -151,3 +151,12 @@ def test_migrate_report_event(self):
...
         self.assertIsInstance(re_5.variantClassification, new_model.VariantClassification)
+
+    def test_migrate_clinical_report_rd(self, fill_nullables=True):
+        cr_rd_6 = self.get_valid_object(object_type=old_model.ClinicalReport, version=self.version_7_0, fill_nullables=fill_nullables)
+        cr_rd_5 = MigrateReports600To500().migrate_clinical_report_rd(old_instance=cr_rd_6)
+        self.assertIsInstance(cr_rd_5, new_model.ClinicalReportRD)
+        self.assertTrue(cr_rd_5.validate(cr_rd_5.toJsonDict()))
+
+    def test_migrate_clinical_report_rd_no_nullables(self):
+        self.test_migrate_clinical_report_rd(fill_nullables=False)
```
Comment (greglever) [inline protocols/tests/test_migration/test_migration_reports_6_0_0_to_reports_5_0_0.py]: good spot @priesgo

File: protocols/migration/migration_reports_6_0_0_to_reports_5_0_0.py (hunk 2 above)
Comment (priesgo) [reply protocols/migration/migration_reports_6_0_0_to_reports_5_0_0.py]: yes

File: protocols/tests/test_migration/test_migration_reports_6_0_0_to_reports_5_0_0.py (hunk 3 above)
Comment (priesgo) [reply protocols/tests/test_migration/test_migration_reports_6_0_0_to_reports_5_0_0.py]: nice

**Comment by greglever** [comment]
Thanks for the review @priesgo 
Thoughts @kevinpetersavage ?

**Comment by greglever** [comment]
@kevinpetersavage -- any idea how to get Travis to run again, the build has passed but it's still displaying as in progress...

**Pull request merged**